components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIProviderHostObject.java

```diff
@@ -1239,7 +1240,8 @@ public static boolean jsFunction_addAPI(Context cx, Scriptable thisObj, Object[]
         }
 
         if (visibility != null && visibility.equals(APIConstants.API_RESTRICTED_VISIBILITY)) {
-            visibleRoles = (String) apiData.get("visibleRoles", apiData);
+            visibleRoles = ((String) apiData.get("visibleRoles", apiData));
```

Comment:
```suggestion
            visibleRoles = (String) apiData.get("visibleRoles", apiData);
```
Remove the extra parentheses